Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue7167.xaml.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Collections.Specialized;
+using System.Linq;
+using System.Windows.Input;
+using Xamarin.Forms.CustomAttributes;
+using Xamarin.Forms.Internals;
+using Xamarin.Forms.Xaml;
+#if UITEST && __WINDOWS__
+using Xamarin.UITest;
+using Xamarin.Forms.Core.UITests;
+using NUnit.Framework;
+#endif
+
+
+namespace Xamarin.Forms.Controls.Issues
+{ 
+#if APP
+	[XamlCompilation(XamlCompilationOptions.Compile)]
+#endif
+
+	[Preserve(AllMembers = true)]
+	[Issue(IssueTracker.Github, 7167,
+		"[Bug] improved observablecollection. a lot of collectionchanges. a reset is sent and listview scrolls to the top", PlatformAffected.UWP)]
+	public partial class Issue7167 : TestContentPage
+	{
+
+		protected override void Init()
+		{
+#if APP
+			InitializeComponent();
+#endif
+			BindingContext = new Issue7167ViewModel();
+		}
+
+		void MyListView_OnItemSelected(object sender, SelectedItemChangedEventArgs e)
+		{
+			var item = e.SelectedItem;
+			var index = e.SelectedItemIndex;
+
+		}
+
+#if UITEST && __WINDOWS__
+		const string ListViewId = "ListViewId";
+		const string AddCommandID = "AddCommandID";
+		const string ClearListCommandId = "ClearListCommandId";
+		const string AddRangeCommandId = "AddRangeCommandId";
+		const string AddRangeWithCleanCommandId = "AddRangeWithCleanCommandId";
+
+		[Test]
+		public  void Issue7167Test()
+		{
+			// arrange
+			// add items to the list and scroll down till item "25"
+			RunningApp.Screenshot("Empty ListView");
+			RunningApp.Tap(q => q.Button(AddRangeCommandId));
+			RunningApp.Tap(q => q.Button(AddRangeCommandId));
+			RunningApp.WaitForElement(c => c.Index(25).Property("Enabled", true));
+			RunningApp.Print.Tree();
+			RunningApp.ScrollDownTo(a => a.Marked("25").Property("text").Contains("25"),
+				b => b.Marked(ListViewId), ScrollStrategy.Auto);
+			RunningApp.WaitForElement(x => x.Marked("25"));
+
+			// act
+			// when adding additional items via a addrange and a CollectionChangedEventArgs.Action.Reset is sent
+			// then the listview shouldnt reset or it should not scroll to the top
+			RunningApp.Tap(q => q.Marked(AddRangeCommandId));
+
+			// assert
+			// assert that item "25" is still visible
+			var result = RunningApp.Query(x => x.Marked(ListViewId).Child().Marked("25"));
+			Assert.That(result?.Length <= 0);
+		}
+#endif
+
+		
+	}
+
+	[Preserve (AllMembers = true)]
+	internal class Issue7167ViewModel
+	{
+		//private static readonly IEnumerable<string> _items = Enumerable.Range(0, 50).Select(num => num.ToString());
```

Comment:
Do we need this?

Reply:
Thanks for response. :)
Do you mean the ViewModel or the commented code?
I guess the comment?

Reply:
Yeah, just the commented code. We like to remove it if it's not necessary :)

Reply:
Oh ok, ye its already removed with the last commit. 😊

Reply:
Thanks! Okay, I think that's the last of the code changes. We will now test this against our regression suite, and if that goes well, it'll be merged! Thanks so much!!